adr/2024-09-states.md

```diff
@@ -0,0 +1,76 @@
+
+#### Separate file - different keys
+
+In this option the text for state specific copy would be stored together. The `/app/i18n/locales/en/translation.json` file would have keys like: `add_expense_amount_field` and then state specific overrides like: `add_expense_amount_field__or` where "or" is the state key for Oregon. The `t()` function would keep track of which state configuration is being used and look up the state specific key if that doesn't exist it would fallback to the regular key.
+
+#### Separate files - same key
+
+In this option the text for the state specific copy is stored in spearate files such as `/app/i18n/locales/en/or.json`. This file would include keys such as `add_expense_amount_field` that overwrite the values in the `/app/i18n/locales/en/translation.json` file. The `t()` function would first look to the state configuration file before looking to the fallback file. This functionality does not exist in the i18n next infrastructure and would need to be developed.
+
+### Routing between configurations
+
+Once an end user arrives on iv-verify they need to be able to start an application for a particular state. Which state a user is using can be configured in either the URL or in a Cookie.
+
+#### URL
+
+In this option the URL for the ledger add screen goes from https://verify-prototype.app.cloud.gov/ledger/income/add to https://verify-prototype.app.cloud.gov/or/ledger/income/add where "or" is the state configuration key (OR for Oregon). This option makes it very clear which flow you are in and can make testing easier. The downside is that it requires a bit of work to modify the URLRouter to generate the URLs correctly to navigate between screens.
+
+#### Cookie
+
+In this option the URL for the ledger add screen states the same but when the user initiates the flow a cookie is set to indicate which flow the user is in. Both the client and server will have access to this cookie to navigate the user down the flow. The downside to this solution is that is not visibly clear which flow a user is unless the UI is modified to say so.
+
+## Decision
+
+The recommendations are as follows:
+
+### Storage
+
+Configuration should be stored in files for iv-verify. This requires the least upfront work and provides version control for configuration changes. iv-verify is not at the point where making configuration changes without a deploy is necessary.
+
+### Routing
+
+Routing should be handled through the URL. This option may require additional up front work as the URLs will need to be altered but it does create a clear system for indicating which configuration is being used. As iv-verify currently utalizes a high degree of static rendering which can be easier done on by URL.
```

Comment:
agree here as well, though a downside of this is that users could change the URL themselves or get themselves lost.